### PR TITLE
feat: automatically detect port number when using the appium-inspector-plugin

### PR DESCRIPTION
**Pull request by wswebcreation**
This pull request adds a new feature to automatically extract and set the server port from the browser URL when running in plugin mode, ensuring the inspector connects to the correct port. It also integrates this logic into the session initialization flow and provides unit tests to verify the new behavior.

**Feature: Port Extraction from URL**
* Added `setPortFromUrl` action in `SessionBuilder.js` to extract the port from the browser URL and set it in the server state, prioritizing plugin mode scenarios.

**Integration: Session Initialization**
* Registered and invoked `setPortFromUrl` in the `SessionBuilder.jsx` component to ensure the port is set during session initialization.

**Testing: Unit Coverage**
* Created `session-builder-setPortFromUrl.spec.js` with tests covering valid port extraction, default port scenarios, absence of port, invalid port values, and edge cases for browser/electron environments.

### CODE REVIEW

**Comment by wswebcreation** [comment]
@eglitise 

Love to get your feedback, this is based on the `wdio-appium-service` feature. I saw that the port was not picked up and this makes it a bit easier for people

File: app/common/renderer/actions/SessionBuilder.js [hunk 1]
```diff
@@ -953,6 +953,33 @@ export function setAddVendorPrefixes(addVendorPrefixes) {
...
+    }
+
+    try {
+      const url = new URL(window.location.href);
+      const port = url.port;
+
+      if (port && url.pathname.includes('/inspector')) {
+        const parsedPort = parseInt(port, 10);
+        if (!isNaN(parsedPort) && parsedPort > 0 && parsedPort <= 65535) {
+          setServerParam('port', parsedPort.toString(), SERVER_TYPES.REMOTE)(dispatch, getState);
```
[the remark this answers is not in the document]
Comment (KazuCocoa) [reply app/common/renderer/actions/SessionBuilder.js]: https://github.com/appium/appium-inspector/blob/56d69f52a538dba01b41e36199c50ba707ebe6f6/app/common/renderer/actions/SessionBuilder.js#L526-L532

**Review by KazuCocoa — APPROVED** [review]
